CoastalResilienceNetwork/GeositeFramework
# Add tool to expand and shrink the map

## Overview

Add button to the map that allows the user to expand and shrink the size of the map by collapsing the sidebar.

This is also done automatically at small screen sizes when plugins are activated, so the tool listens for changes to the map size so that the icon can be appropriately updated (or else it would become out of sync).

Since the CSS rules for the collapsed sidebar were in a media query, they are copied and placed outside of the media query as well so that they can be used at all screen sizes. This is not ideal.

Connects #887 

### Demo

![tnc887](https://cloud.githubusercontent.com/assets/1042475/23562247/acb02f1c-000f-11e7-9fab-5ac94f5d7115.gif)

## Testing Instructions

- Without any active plugins, use the button to expand and shrink the map. Verify that the button does indeed do this, and that the icon of the button stays in sync with the changes.
- Reduce the size of your browser window to less than 990 pixels.
- Activate and minimize plugins, and verify that the map expand tool button icon stays in sync with the changes.
